File: Data Structures & Algorithms/max-water-container/submission-6.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        area = 0
        i = 0
        j = len(heights) - 1
        while i != j:
            area = max(area, self.Area(i, j, heights[i], heights[j]))
            if heights[i] > heights[j]:
                j -= 1
            else:
                i += 1
        return area
    def Area(self, idx, idx2, h_1, h_2):    
        return (idx2 - idx) * (min(h_1, h_2))

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-6.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        std::vector<int> dp;
        dp = std::vector<int>(n, 1);

        for (int i = n - 1; i >= 0; i--) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] < nums[j]) {
                    dp[i] = std::max(dp[i], 1 + dp[j]);
                }
            }
        }
        return std::ranges::max(dp);
    }
};
